keel-core/src/main/kotlin/com/netflix/spinnaker/keel/services/ApplicationService.kt

```diff
@@ -552,25 +553,52 @@ class ApplicationService(
     val releaseStatus = repository.getReleaseStatus(artifact, version)
     return repository.getArtifactVersion(artifact, version, releaseStatus)
   }
-}
 
-/**
- * Query the repository for all of the verification states associated with [versions]
- *
- * This just calls [KeelRepository.getVerificationStatesBatch] and reshapes the returned value to a map
- */
-fun KeelRepository.getVerificationStates(
```

Comment:
This function is a bit of a hairball. I made an effort ti refactor it into smaller functions.